repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/intrafilterlogging/IntraFilterLoggingTest.groovy

```diff
@@ -126,13 +126,18 @@ class IntraFilterLoggingTest extends ReposeValveTest {
                 ], json)
             }
         }
+
+        // todo: fails because the x-trans-id header is not logged on responses
+        // todo: this seems to be caused by the response wrapper not exposing the headers of the wrapped response
+        // todo: since we set the x-trans-id header up front, it is not visible to the intrafilter logging mechanism. but it does make it back to the client
+        // todo: this was not failing prior due to the UUID header being added after returning from doFilter (just prior to logging)
```

Comment:
Just wanted to call this out. The way I see it, we have two ways to fix this:
1. Update the `ReposeFilter` to add the `X-Trans-Id` header on the way out as we were doing before.
2. Update the `HttpServletResponseWrapper` to expose the headers on the wrapped response.

I should add that the reason we are not currently exposing the headers on the wrapped response is that we have no way of removing headers on the wrapped response. The `HttpServletResponse` interface simply does not allow it. In the end, we did not want to expose headers to those using our wrapper (via calls like `getHeaderNames` and `getHeaders`) that could not be modified in the same way as every other header.

Of course, we could use the `setHeader` method to remove headers in some cases. The Jetty response class, for example, will remove a header if `setHeader(..., null)` is called. However, we have no guarantee that other wrappers will work the same way. 

The compromise might be to expose headers from the wrapped response, and make a best effort at removing them if `removeHeader` is called, logging a warning if we were unable to remove the header (which is easily checked by querying the wrapped response after calling `setHeader`).

Reply:
When were we applying it previously?

Reply:
Here:
https://github.com/rackerlabs/repose/blob/master/repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilterChain.java#L242-L247

As you can see, we used to pass the ID from the request and use that ID if the response did not have one. Well, I suspect the response never had one, and so the request ID was always being used. Correct outcome, but not the best way of getting there. Especially considering that it was masking the issue I described above, which is a general issue.

Reply:
I'd say slap it on as soon as we get a response from the origin service